nuxeo-compound-documents/src/test/java/org/nuxeo/compound/documents/TestCompoundDocument.java

```diff
@@ -16,65 +16,109 @@
-        RenderingContext ctx = RenderingContext.CtxBuilder.enrichDoc("subtypes").get();
-        JsonAssert json = jsonAssert(doc, ctx);
-        json = json.has("contextParameters").isObject();
-        json.properties(1);
-        json = json.has("subtypes").isArray();
-        json.childrenContains("type", "File", "Picture", "Video", "Audio", "CompoundDocumentFolder");
+        var formData = new MultivaluedMapImpl();
+        formData.putSingle("doctype", "CompoundDocument");
+        formData.putSingle("name", "myCompoundDocument");
+        try (CloseableClientResponse response = httpClientRule.post("/path/", formData)) {
+            assertEquals(SC_SEE_OTHER, response.getStatus());
+        }
+        try (CloseableClientResponse response = httpClientRule.get(
+                "/path/myCompoundDocument?" + MarshallingConstants.EMBED_ENRICHERS + ".document=subtypes")) {
+            checkSubtypes(response, "CompoundDocument");
+        }
+        createSubTypes("myCompoundDocument");
+        try (CloseableClientResponse response = httpClientRule.get("/path/myCompoundDocument/myCompoundDocumentFolder?"
+                + MarshallingConstants.EMBED_ENRICHERS + ".document=subtypes")) {
+            checkSubtypes(response, "Folderish");
+        }
+        createSubTypes("myCompoundDocument/myCompoundDocumentFolder");
+    }
+
+    protected void createSubTypes(String docPath) {
+        MultivaluedMap<String, String> formData = new MultivaluedMapImpl();
+        subtypes.forEach(subtype -> {
+            formData.putSingle("doctype", subtype);
+            formData.putSingle("name", "my" + subtype);
+            try (CloseableClientResponse response = httpClientRule.post("/path/" + docPath, formData)) {
+                assertEquals(SC_SEE_OTHER, response.getStatus());
+            }
+        });
+    }
 
-        doc = session.createDocumentModel("/", "test", "CompoundDocumentFolder");
-        assertFalse(doc.getFacets().contains("CompoundDocument"));
-        assertTrue(doc.getFacets().contains("Folderish"));
-        ctx = RenderingContext.CtxBuilder.enrichDoc("subtypes").get();
-        json = jsonAssert(doc, ctx);
-        json = json.has("contextParameters").isObject();
-        json.properties(1);
-        json = json.has("subtypes").isArray();
-        json.childrenContains("type", "File", "Picture", "Video", "Audio", "CompoundDocumentFolder");
+    protected void checkSubtypes(CloseableClientResponse response, String expectedFacet) throws IOException {
+        assertEquals(SC_OK, response.getStatus());
+        JsonNode node = mapper.readTree(response.getEntityInputStream());
+        assertEquals(1, node.get("facets").size());
+        assertEquals(expectedFacet, node.get("facets").get(0).asText());
+        assertEquals(5, node.get("contextParameters").get("subtypes").size());
+        node.get("contextParameters")
+            .get("subtypes")
+            .forEach(t -> assertTrue(subtypes.contains(t.get("type").asText())));
```

Comment:
You should do assertions like you did for facets, because with the forEach way there could be no subtypes in the response and your assertion will succeed.

Reply:
This is why I check the number of facets just before. Not enough ?

Reply:
You should do the same than facets on `subtypes`.

Reply:
continued on slack. 
I meant subtypes and not facets 😅 .
We agreed also it was ok if we are checking how much subtypes there are in total.